### PR TITLE
Fix header for function 'now'

### LINKED ISSUE #7100: clickhouse benchmark <<< "SELECT now()" does not work on master

### PR DESCRIPTION
**Pull request by KochetovNicolai**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

For changelog. Remove if this is non-significant change.

Category (leave one):
- Bug Fix

Short description (up to few sentences):
Fix exception `Cannot convert column ... because it is constant but values of constants are different in source and result` which could rarely happen when functions `now()`, `today()`, `yestarday()`, `randConstant()` are used.

Detailed description (optional):
Changed in `ExpressionAnalyzer`: `ExpressionActions` from `getSampleBlock` are used now in `readImpl`. Also value for  functions `now()`, `today()`, `yestarday()`, `randConstant()` is determined in `FunctionBuilder`. Fixes #7100.

### CODE REVIEW

**Comment by alexey-milovidov** [comment]
We have more than one such function. What about today(), yesterday(), randConstant()...

**Review by alexey-milovidov — CHANGES_REQUESTED** [review]
.

**Comment by KochetovNicolai** [comment]
Still doesn't fix :(

**Comment by KochetovNicolai** [comment]
Now won't fail.
Can't solve it better for now.

**Comment by alexey-milovidov** [comment]
Missing test.

File: dbms/src/DataStreams/ConvertingBlockInputStream.cpp [hunk 1]
```diff
@@ -34,8 +34,10 @@ ConvertingBlockInputStream::ConvertingBlockInputStream(
     const Context & context_,
     const BlockInputStreamPtr & input,
     const Block & result_header,
-    MatchColumnsMode mode)
+    MatchColumnsMode mode,
+    bool allow_different_constant_values_)
```
Comment (alexey-milovidov) [inline dbms/src/DataStreams/ConvertingBlockInputStream.cpp]: A huge complication. Better to find other way to fix.

File: dbms/src/DataStreams/ConvertingBlockInputStream.h [hunk 2]
```diff
@@ -49,6 +50,8 @@ class ConvertingBlockInputStream : public IBlockInputStream
     /// How to construct result block. Position in source block, where to get each column.
     using Conversion = std::vector<size_t>;
     Conversion conversion;
+
+    bool allow_different_constant_values;
```
Comment (alexey-milovidov) [inline dbms/src/DataStreams/ConvertingBlockInputStream.h]: You have not explained it in code.

**Comment by alexey-milovidov** [comment]
> today(), yesterday(), randConstant()

I don't think I have listed all the relevant functions.

**Comment by KochetovNicolai** [comment]
Now actions from header are reused. Should fix for all functions.

**Review by alexey-milovidov — APPROVED** [review]

**Pull request merged**